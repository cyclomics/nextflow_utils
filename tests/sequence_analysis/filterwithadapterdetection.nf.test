nextflow_workflow {

  name "Test FilterWithAdapterDetection"
  script "sequence_analysis/subworkflows.nf"
  workflow "FilterWithAdapterDetection"

  test("Check if read filtering is correct") {

    when {
      workflow {
        """
        sample_id = "SOME_ID_1"
        file_id = "SOME_ID_2"
        fq_file = file("$baseDir/example_datasets/example_native_reads/fastq_pass/FAW79986_pass_dbcafcd9_40623079_0.fastq.gz")

        input[0] = Channel.of([sample_id, file_id, fq_file])
        """
      }
    }

    then {
      assert workflow.success
      assert snapshot(workflow.out).match()
    }
  }

}