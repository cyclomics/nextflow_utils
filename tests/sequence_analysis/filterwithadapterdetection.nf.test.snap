{
    "Check if read filtering is correct": {
        "content": [
            {
                "0": [
                    [
                        "SOME_ID_1",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.fastq:md5,b43c6c664b0d8a96f1c3943bcbdf33e8"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-02-05T15:58:22.106656"
    }
}